lib/winpki/locate.go

```diff
@@ -0,0 +1,69 @@
+/*
+ * Teleport
+ * Copyright (C) 2025  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package winpki
+
+import (
+	"context"
+	"net"
+	"os"
+
+	"github.com/gravitational/trace"
+)
+
+// LocateLDAPServer looks up the LDAP server in an Active Directory
+// environment by implementing the DNS-based discovery DC locator
+// process.
+//
+// See https://learn.microsoft.com/en-us/windows-server/identity/ad-ds/manage/dc-locator?tabs=dns-based-discovery
+func LocateLDAPServer(ctx context.Context, domain string, site string, resolver *net.Resolver) ([]string, error) {
+	var finalDomain = domain
```

Comment:
I would avoid creating two variables that hold the domain, because then it would be easy to accidentally use the wrong one.

How about this?

```
if site != "" {
    domain = site + "._sites." + domain
}
```